File: Challenges/Javascript/nextLarger.js
function nextLarger(a) {
    let out = [];

    for(let x = -1, i = 0, f = 0; i < a.length; i++, f = 0) {
        for(let j = i; j < a.length; j++) {
            if(a[j] > a[i]) {
                f = a[j];
                break;
            }
        }
        out.push(f > 0 ? f : -1);
    }

    return out;
}

File: Challenges/Java/NextLarger.java
public final class NextLarger {

	int[] nextLarger(int[] a) {
	    int l = a.length;
	    int[] out = new int[l];

	    for(int x = -1, i = 0, f = 0; i < l; i++, f = 0) {
	        for(int j = i; j < l; j++) {
	            if(a[j] > a[i]) {
	                f = a[j];
	                break;
	            }
	        }
	        out[x += 1] = f > 0 ? f : -1;
	    }

	    return out;
	}
}